app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -156,17 +200,76 @@ public void updateIfNeeded() {
             changed = true;
         }
 
+        /* Section - options */
         // Disable weak hash benchmark for faster startup.
         // https://github.com/syncthing/syncthing/issues/4348
         Element options = (Element) mConfig.getDocumentElement()
                 .getElementsByTagName("options").item(0);
         changed = setConfigElement(options, "weakHashSelectionMethod", "never") || changed;
 
+        /* Check if we have to dismiss any specific "unackedNotificationID" */
+        /* Dismiss "fsWatcherNotification" according to https://github.com/syncthing/syncthing-android/pull/1051 */
+        if (getConfigElement(options, "unackedNotificationID").contains("fsWatcherNotification")) {
+            Log.i(TAG, "Remove 'fsWatcherNotification' from 'unackedNotificationID' node.");
+            changed = setConfigElement(options, "unackedNotificationID", getConfigElement(options, "unackedNotificationID").replace("fsWatcherNotification", "")) || changed;
+        }
+
+        // Save changes if we made any.
         if (changed) {
             saveChanges();
         }
     }
 
+    /**
+     * Updates syncthing options to a version specific target setting in the config file.
+     *
+     * Used for one-time config migration from a lower syncthing version to the current version.
+     * Enables filesystem watcher.
+     */
+    private boolean migrateSyncthingOptions () {
+        boolean changed = false;
+
+        /* Read existing config version */
+        int iConfigVersion = Integer.parseInt(mConfig.getDocumentElement().getAttribute("version"));
+        Log.i(TAG, "Found existing config version " + Integer.toString(iConfigVersion));
+
+        /* Check if we have to do manual migration from version X to Y */
+        if (iConfigVersion == 27) {
+            /* fsWatcher transition - https://github.com/syncthing/syncthing/issues/4882 */
+            Log.i(TAG, "Migrating config version " + Integer.toString(iConfigVersion) + " to 28 ...");
+
+            /* Enable fsWatcher for all folders */
+            NodeList folders = mConfig.getDocumentElement().getElementsByTagName("folder");
+            for (int i = 0; i < folders.getLength(); i++) {
+                Element r = (Element) folders.item(i);
+
+                // Enable "fsWatcherEnabled" attribute.
+                Log.i(TAG, "Set 'fsWatcherEnabled' on folder " + r.getAttribute("id"));
+                r.setAttribute("fsWatcherEnabled", Boolean.toString(true));
+                changed = true;
+            }
+
+            /**
+            * Set config version to 28 after manual config migration
+            * This prevents "unackedNotificationID" getting populated
+            * with the fsWatcher GUI notification.
+            */
+            iConfigVersion = 28;
+            mConfig.getDocumentElement().setAttribute("version", Integer.toString(iConfigVersion));
+            Log.i(TAG, "Config version " + Integer.toString(iConfigVersion) + " reached.");
```

Comment:
Probably want to move the attribute setting and logging juist before the return, as there might be more than one migration, so no point setting the version attribute in each migration, as you can just adjust the stack variable in each individual migration.

Reply:
Good thought :)

Reply:
Also found a problem and fixed it, too. "changed = true" must not live in the for. Assume no folder would have changed because a user enabled fsWatcher manually before the migration, the config would then not have been written back because changed would still have been "false" beside we bumped the version number.